packages/block-editor/src/hooks/duotone.js

```diff
@@ -120,19 +122,30 @@ ${ selector } {
 	);
 }
 
-function DuotonePanel( { attributes, setAttributes } ) {
+function DuotonePanel( { name, attributes, setAttributes } ) {
 	const style = attributes?.style;
 	const duotone = style?.color?.duotone;
 
-	const duotonePalette = useSetting( 'color.duotone' );
-	const colorPalette = useSetting( 'color.palette' );
+	const duotonePalette = useSetting( 'color.duotone' ) || EMPTY_ARRAY;
+	const colorPalette = useSetting( 'color.palette' ) || EMPTY_ARRAY;
 	const disableCustomColors = ! useSetting( 'color.custom' );
+	const disableCustomDuotone =
+		! useSetting( 'color.customDuotone' ) ||
+		( colorPalette?.length === 0 && disableCustomColors );
```

Comment:
I would expect this check to go [inside `DuotonePicker`](https://github.com/WordPress/gutenberg/pull/33295/files#diff-37be335b0d340015a545c960c96de07c7e0ea34ce12b3dd8a1735930a526525aR82-R85) where the `colorPalette` is actually used.

Reply:
It's a few different things: within the component, we need to make sure the individual subcomponents are visible/hidden when correspond, and we do that. We also need to check that the whole component is hidden when all subparts are also hidden: we need to check for that at this level, before the component is rendered.